### PR TITLE
Explore: resize graph on window resize

### LINKED ISSUE #13517: Explore: graph doesn't resize when I resize window
screen shot 2018-10-04 at 07 56 58

### PR DESCRIPTION
**Pull request by davkal**
Fixes #13517

### CODE REVIEW

**Comment by torkelo** [comment]
The react dashboard grid uses a sizeMe lib, not sure that could be useful in this scenario

**Comment by davkal** [comment]
I find the libs quite bloated. Some are useful because they polyfill the window object needed for SSR. But we'd need to revisit the whole code anyway when introducing that.

**Comment by torkelo** [comment]
> I find the libs quite bloated. 

sizeMe is bloated? In what way? looks very small an lean and isolates this interaction so much more cleanly.  A lot better than manually hooking up events to window directly from the component and having to unsub on component did unmount. 

Also it handles cases that this solution does not, like hiding the sidebar (will not cause a resize event) but the available space for graph will increase.

**Comment by torkelo** [comment]
I mean the Graph could be simplified by wrapping it in sizeMe HOC that adds width, then you dont need to use jquery to calculate width in the draw function

**Comment by davkal** [comment]
Good points, I'll give it a try.

File: public/app/features/explore/Graph.tsx [hunk 1]
```diff
@@ -68,7 +69,7 @@ const FLOT_OPTIONS = {
   // },
 };
 
-class Graph extends Component<any, any> {
+class Graph extends Component<any, { showAllTimeSeries: boolean }> {
```
Comment (peterholmberg) [inline public/app/features/explore/Graph.tsx]: extract to interface?

```
interface State {
 showAllTimeSeries: boolean;
}
```

**Review by torkelo — APPROVED** [review]
nice!

**Pull request merged**